### PR TITLE
styled points picker added

### PR DESCRIPTION
**Pull request by shivanshu877**
### Information

- Jira: YT-12200
- Link of PR pointing to `@develop`: 
  
### I've done the following things before and after raising the PR

* Put the JIRA number `[YT-NNNN]` in the PR title.
* Done a self review of the PR.
* Assigned reviewers to this PR.

### CODE REVIEW

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import 'package:flutter/material.dart';
+import 'package:yc_app_utils/models/form_component/styled_points_picker.model.dart';
+import 'package:yc_app_utils/yc_app_utils.dart';
+
+class PointsPickerWidget extends StatefulWidget {
+  final StyledPointsPickerModel? pointsPickerdata;
+  final ValueChanged<int?> onItemSelected;
+  const PointsPickerWidget(
+      {super.key, this.pointsPickerdata, required this.onItemSelected});
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:9]:
```suggestion
      {super.key, this.pointsPickerdata, required this.onItemSelected,});
```

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  void _selectItem(int index) {
+    _selectedIndexNotifier.value = index;
+    widget.onItemSelected(index);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final options = widget.pointsPickerdata?.pickerOptions ?? [];
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:38]:
```suggestion
   type  get options => widget.pointsPickerdata?.pickerOptions ?? [];
```

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            children: [
+              ValueListenableBuilder<int?>(
+                valueListenable: _selectedIndexNotifier,
+                builder: (context, selectedIndex, child) {
+                  return Wrap(
+                    spacing: 12.0,
+                    runSpacing: 16.0,
+                    alignment: WrapAlignment.center,
+                    children: List.generate(options.length, (index) {
+                      final entry = options[index];
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:64]:
```suggestion
                      final pickerOption = options[index];
```

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                              color: selectedIndex == index
+                                  ? Colors.white
+                                  : AppColors.parseStringToColor(
+                                      entry.borderColor),
+                            ),
+                          ),
+                          child: V2StyledTextWidget(
+                            styledText: selectedIndex == index
+                                ? entry.selectedComponent!
+                                : entry.unSelectedComponent!,
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:84]:
```suggestion
                            styledText: selectedIndex == index
                                ? entry.selectedComponent!
                                : entry.unSelectedComponent!,
```

add null checks and only then show

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                    }),
+                  );
+                },
+              ),
+            ],
+          ),
+        ),
+        if (widget.pointsPickerdata?.bottomLabel != null)
+          Padding(
+            padding: const EdgeInsets.only(right: 16.0),
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:97]:
```suggestion
            padding: const EdgeInsets.only(right: AppSpacing.x),
```

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            padding: const EdgeInsets.only(left: 16.0),
+            child: Align(
+              alignment: widget.pointsPickerdata!.topLabelAlignment ??
+                  Alignment.centerLeft,
+              child: V2StyledTextWidget(
+                  styledText: widget.pointsPickerdata!.topLabel!),
+            ),
+          ),
+        Padding(
+          padding: const EdgeInsets.all(12.0),
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:52]:
```suggestion
          padding: const EdgeInsets.all(AppSpacing.x),
```

**Review by naman-jn — CHANGES_REQUESTED** [review]

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+                  styledText: widget.pointsPickerdata!.bottomLabel!),
+            ),
+          ),
+      ],
+    );
+  }
+}
+
+Widget getPickerOptions(CircularButton pickerOption, int index,
+    int? selectedIndex, ValueChanged<int> onItemSelected) {
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:95]:
```suggestion
    int? selectedIndex, ValueChanged<int> onItemSelected,) {
```
convert to named parameters

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+            children: [
+              ValueListenableBuilder<int?>(
+                valueListenable: _selectedIndexNotifier,
+                builder: (context, selectedIndex, child) {
+                  return Wrap(
+                    spacing: 12.0,
+                    runSpacing: 16.0,
+                    alignment: WrapAlignment.center,
+                    children: List.generate(options.length, (index) {
+                      final pickerOption = options[index];
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:69]:
```suggestion
                      final pickerOption = options[index];
                      bool isSelected = selectedIndex == index;
```

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+                      return getPickerOptions(
+                          pickerOption, index, selectedIndex, _selectItem);
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:71]:
```suggestion
                          pickerOption, index, isSelected , _selectItem);
```

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+    return Column(
+      children: [
+        if (widget.pointsPickerdata?.topLabel != null)
+          Padding(
+            padding: const EdgeInsets.only(left: 16.0),
+            child: Align(
+              alignment: widget.pointsPickerdata!.topLabelAlignment ??
+                  Alignment.centerLeft,
+              child: V2StyledTextWidget(
+                  styledText: widget.pointsPickerdata!.topLabel!),
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:53]:
```suggestion
                  styledText: widget.pointsPickerdata!.topLabel!,),
```

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+  List<CircularButton> get options =>
+      widget.pointsPickerdata?.pickerOptions ?? [];
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        if (widget.pointsPickerdata?.topLabel != null)
+          Padding(
+            padding: const EdgeInsets.only(left: 16.0),
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart:48]:
```suggestion
            padding: const EdgeInsets.only(AppSpacing.x),
```

**Review by naman-jn — CHANGES_REQUESTED** [review]

File: lib/ui/components/form_components/styled_points_picker.widget.dart (hunk 11 above)
Comment (shivanshu877) [reply lib/ui/components/form_components/styled_points_picker.widget.dart]: done

File: lib/ui/components/form_components/styled_points_picker.widget.dart (hunk 10 above)
Comment (shivanshu877) [reply lib/ui/components/form_components/styled_points_picker.widget.dart]: added

File: lib/ui/components/form_components/styled_points_picker.widget.dart (hunk 9 above)
Comment (shivanshu877) [reply lib/ui/components/form_components/styled_points_picker.widget.dart]: done

File: lib/ui/components/form_components/styled_points_picker.widget.dart (hunk 8 above)
Comment (shivanshu877) [reply lib/ui/components/form_components/styled_points_picker.widget.dart:70]: done

File: lib/ui/components/form_components/styled_points_picker.widget.dart (hunk 6 above)
Comment (shivanshu877) [reply lib/ui/components/form_components/styled_points_picker.widget.dart]: done

File: lib/ui/components/form_components/styled_points_picker.widget.dart (hunk 5 above)
Comment (shivanshu877) [reply lib/ui/components/form_components/styled_points_picker.widget.dart]: done

File: lib/ui/components/form_components/styled_points_picker.widget.dart (hunk 3 above)
Comment (shivanshu877) [reply lib/ui/components/form_components/styled_points_picker.widget.dart]: done

File: lib/ui/components/form_components/styled_points_picker.widget.dart (hunk 2 above)
Comment (shivanshu877) [reply lib/ui/components/form_components/styled_points_picker.widget.dart]: done

File: lib/ui/components/form_components/styled_points_picker.widget.dart (hunk 1 above)
Comment (shivanshu877) [reply lib/ui/components/form_components/styled_points_picker.widget.dart]: done

File: lib/ui/components/form_components/styled_points_picker.widget.dart [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+      ],
+    );
+  }
+}
+
+Widget getPickerOptions(
+  CircularButton pickerOption,
+  int index,
+  bool isSelected,
+  ValueChanged<int> onItemSelected,
```
Comment (naman-jn) [inline lib/ui/components/form_components/styled_points_picker.widget.dart]:
```suggestion
  CircularButton pickerOption,
  int index,
  bool isSelected,
  ValueChanged<int> onItemSelected,
```

convert to named params

**Review by naman-jn — CHANGES_REQUESTED** [review]

**Review by naman-jn — APPROVED** [review]

**Pull request merged**